### PR TITLE
SFO 69: Unit test for register feature

### PR DESCRIPTION
**Pull request by josh231101**
Unit test for register feature

### CODE REVIEW

File: josue/app.test.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    });
+    it('should save to localStorage',()=>{
+        const dummyUser = {"name":"Juan Lopez","phone":"8332041352","email":"juanperez@gmail.com","password":"12345","password_confirmation":"12345"}
+        localStorage.setItem('8332041352', dummyUser);
+        expect(localStorage.setItem).toHaveBeenLastCalledWith('8332041352',dummyUser);
+        console.log('++localstorage: ',localStorage)
+    });
+
+    it('renders Sign In form',()=>{
+        expect(container.querySelector('#form')).toBeInTheDocument();
```
Comment (frank-martinez-iest) [inline josue/app.test.js:35]: Add this test below this line: `expect(container).toMatchSnapshot();`

File: josue/app.test.js [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+       await fireEvent.click(loginBtn);
+
+       const errorMessageElement = await container.querySelector('.error-message');
+       expect(errorMessageElement.innerText).not.toBe('')
+    });
+    it('must not contain register fields',()=>{
+        /* REGISTER FIELDS DIV*/
+        expect(container.querySelector('#register-fields')).not.toBeInTheDocument();
+    });
+    it('updates submit btn when register btn gets clicked', async()=>{
```
Comment (frank-martinez-iest) [inline josue/app.test.js]: Does this test asserts the inner text of a button changes depending if you are submitting the form or if you click on the button to show the form? If so, rename this test to be `changes the button's inner text after form was submitted`

File: josue/app.test.js [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+const html = fs.readFileSync(path.resolve(__dirname,'./index.html'),'utf-8');
+
+let dom
+let window
+let document
+let container
+let scriptElement
+
+describe('Unit test login feature',()=>{
```
Comment (frank-martinez-iest) [inline josue/app.test.js:16]: I think the name of this describe should be related to `register` instead of login

Comment (frank-martinez-iest) [inline josue/app.test.js:16]: Can you add a test to assert the error message appears when you fill the form with the wrong `confirm password` value?

**Review by frank-martinez-iest — CHANGES_REQUESTED** [review]
This looks great! Good job on the unit tests! These look very similar to the ones we do in my job!

File: josue/app.test.js (hunk 1 above)
Comment (josh231101) [reply josue/app.test.js:35]: Test passed!

File: josue/app.test.js (hunk 3 above)
Comment (josh231101) [reply josue/app.test.js:16]: I decided to move the register tests to another "describe"

**Pull request merged**